src/sentry/api/exceptions.py

```diff
@@ -1,14 +1,55 @@
 from __future__ import absolute_import
 
+from django.core.urlresolvers import reverse
+from rest_framework import status
 from rest_framework.exceptions import APIException
 
 
 class ResourceDoesNotExist(APIException):
-    status_code = 404
+    status_code = status.HTTP_404_NOT_FOUND
 
 
 class ResourceMoved(APIException):
-    status_code = 301
+    status_code = status.HTTP_301_MOVED_PERMANENTLY
+
+
+class SsoRequired(APIException):
+    status_code = status.HTTP_401_UNAUTHORIZED
+
+    def __init__(self, organization):
+        detail = {
+            'code': 'sso-required',
+            'message': 'Must login via SSO',
+            'extra': {
+                'loginUrl': reverse('sentry-auth-organization', args=[organization.slug]),
+            }
+        }
+        super(SsoRequired, self).__init__(detail=detail)
```

Comment:
How does this look @dcramer ?

Reply:
![guccigang](https://user-images.githubusercontent.com/23610/38338354-1dbed450-381e-11e8-8fcf-adb56a93d245.png)